framework/db/ActiveRecord.php

```diff
@@ -291,12 +303,18 @@ public static function tableName()
      */
     public static function getTableSchema()
     {
-        $schema = static::getDb()->getSchema()->getTableSchema(static::tableName());
-        if ($schema !== null) {
-            return $schema;
-        } else {
-            throw new InvalidConfigException("The table does not exist: " . static::tableName());
+        static $tableSchema = null;
+
+        if ($tableSchema === null) {
+            $schema = static::getDb()->getSchema()->getTableSchema(static::tableName());
```

Comment:
@klimov-paul Don't those two comments contradict eachother?

Schema::getTableSchema() indeed already has an [internal cache](https://github.com/yiisoft/yii2/blob/master/framework/db/Schema.php#L117), so if the table structure were to change in the middle of a request, the changes wouldn't automatically be picked up before this change either. If I see it correctly, this causes no functional changes, it just prevents the overhead of function calls to `getDb()`, `getSchema()`, `getTableSchema` and `tableName`.

If someone wanted to explicitly refresh the schema data, they could still call Schema::getTableSchema($refresh=true).

Or am I misunderstanding something?